nexus/src/app/sagas/switch_port_settings_clear.rs

```diff
@@ -179,7 +179,7 @@ async fn spa_undo_clear_switch_port_settings(
     let log = sagactx.user_data().log();
 
     let port_id: PortId = PortId::from_str(&params.port_name)
-        .map_err(|e| external::Error::internal_error(e))?;
+        .map_err(|e| external::Error::internal_error(e.to_string().as_str()))?;
```

Comment:
maybe change `internal_error`?
<img width="688" alt="image" src="https://github.com/oxidecomputer/omicron/assets/677483/cc668f44-3c83-443e-b86e-5302805aeb93">

Maybe these should accept `impl ToString` ? or `AsRef<str>`

Reply:
I agree that would be a good change, I've found myself often calling `to_string()` only to throw that away immediately, where the internal error variant then allocates a new one. I would prefer to do it in another PR though, since this variant is used _extensively_.

Reply:
https://github.com/oxidecomputer/omicron/issues/4892

Reply:
Sounds good